extensions/common/azure/azure-blob-core/src/main/java/org/eclipse/edc/azure/blob/api/BlobStoreApiImpl.java

```diff
@@ -105,29 +108,25 @@ public byte[] getBlob(String account, String container, String blobName) {
     }
 
     private BlobServiceClient getBlobServiceClient(String accountName) {
-        Objects.requireNonNull(accountName, "accountName");
-
-        if (cache.containsKey(accountName)) {
-            return cache.get(accountName);
+        if (this.accountCache.isAccountInCache(accountName)) {
+            return this.accountCache.getAccount(accountName);
         }
 
         var accountKey = vault.resolveSecret(accountName + "-key1");
-        var endpoint = createEndpoint(accountName);
-
-        var blobServiceClient = accountKey == null ?
-                new BlobServiceClientBuilder().credential(new DefaultAzureCredentialBuilder().build())
-                        .endpoint(endpoint)
-                        .buildClient() :
-                new BlobServiceClientBuilder().credential(createCredential(accountKey, accountName))
-                        .endpoint(endpoint)
-                        .buildClient();
 
-        cache.put(accountName, blobServiceClient);
-        return blobServiceClient;
+        return this.accountCache.saveAccount(blobstoreEndpointTemplate, accountName, accountKey);
     }
 
-    private StorageSharedKeyCredential createCredential(String accountKey, String accountName) {
-        return new StorageSharedKeyCredential(accountName, accountKey);
+    private BlobServiceClient getBlobServiceClient(String accountName, String accountKey) {
+        if (StringUtils.isNullOrBlank(accountKey)) {
+            return getBlobServiceClient(accountName);
+        }
+
+        if (this.accountCache.isAccountInCache(accountName)) {
+            return this.accountCache.getAccount(accountName);
+        }
+
+        return this.accountCache.saveAccount(blobstoreEndpointTemplate, accountName, accountKey);
```

Comment:
this logic should be moved into the cache, because the "storage" thing is something internal to the cache concept, the caller should just request the client with the account name and the account key (that could be null), and the cache will take care of eventually instantiate and store.

Reply:
yeah, I understand. I transfered the account cache logic to its own class to try and respect the srp.
thank you!